source/common/upstream/subset_lb.cc

```diff
@@ -463,7 +463,7 @@ void SubsetLoadBalancer::HostSubsetImpl::update(const HostVector& hosts_added,
       original_host_set_.hostsPerLocality().filter(
           [&predicate](const Host& host) { return predicate(host) && host.healthy(); });
 
-  HostSetImpl::updateHosts(hosts, healthy_hosts, hosts_per_locality, healthy_hosts_per_locality,
+  HostSetImpl::updateHosts(hosts, healthy_hosts, hosts_per_locality, healthy_hosts_per_locality, {},
```

Comment:
I think not passing any LocalityWeights here will end up disabling this feature. It seems sufficient to just pass along `original_host_set_.localityWeights()`, though. That way the underlying load balancers have all the information to make the locality weighted load balancing decisions.